### PR TITLE
[main][feat] Enable whitelist caller

### PR DESCRIPTION
**Pull request by spicysquid168**
## Description
Add ability to allow whitelisted non-EOA to interact with restricted functions.

## Why?
For more composability

### CODE REVIEW

File: contracts/6/protocol/Vault.sol [hunk 1]
```diff
@@ -29,6 +29,8 @@ import "../token/interfaces/IFairLaunch.sol";
 import "../utils/SafeToken.sol";
 import "./WNativeRelayer.sol";
 
+import "hardhat/console.sol";
```
Comment (abraham-alpaca) [inline contracts/6/protocol/Vault.sol:32]: remove console.log

File: contracts/6/protocol/SimpleVaultConfig.sol [hunk 2]
```diff
@@ -110,6 +111,13 @@ contract SimpleVaultConfig is IVaultConfig, OwnableUpgradeSafe {
     });
   }
 
+  /// @dev Set whitelisted callers. Must only be called by the owner.
```
Comment (abraham-alpaca) [inline contracts/6/protocol/SimpleVaultConfig.sol:114]: please include an event for setWhitelistedCallers

File: test/Vault_PancakeswapV2.test.ts [hunk 3]
```diff
@@ -403,67 +424,212 @@ describe('Vault - PancakeswapV2', () => {
...
-          ['address', 'bytes'],
-          [addStrat.address, ethers.utils.defaultAbiCoder.encode(
-            ['uint256'],
-            ['0'])
-          ]
-        ),
-      );
-    });
+    context('when user is contract', async() => {
+      it('should revert if evil contract try to call onlyEOAEx function', async () => {
```
Comment (abraham-alpaca) [inline test/Vault_PancakeswapV2.test.ts:428]: onlyEOAEx => onlyEOAorWhitelisted

File: test/Vault_PancakeswapV2.test.ts [hunk 4]
```diff
@@ -403,67 +424,212 @@ describe('Vault - PancakeswapV2', () => {
...
+                )]
+            )]
+          )
+        )
 
-    it('should not allow to open a position with debt less than MIN_DEBT_SIZE', async () => {
-      // Deployer deposits 3 BTOKEN to the bank
-      await baseToken.approve(vault.address, ethers.utils.parseEther('3'));
-      await vault.deposit(ethers.utils.parseEther('3'));
+        const [worker, owner, ] = await vault.positions(1)
```
Comment (abraham-alpaca) [inline test/Vault_PancakeswapV2.test.ts:483]: no need to comma the last one

**Review by abraham-alpaca — APPROVED** [review]

**Review by sundayalpaca — APPROVED** [review]

**Pull request merged**